### ISSUE TITLE
Usability concern with DiskStorageHealthCheck

### ISSUE THREAD

**Issue by MuiBienCarlota** [Xabaril/AspNetCore.Diagnostics.HealthChecks#375]
When using `AddDiskStorageHealthCheck`, `setup` argument allow us to add several drive limit  checks through `AddDrive`:

``` csharp
healthCheckBuilder.AddDiskStorageHealthCheck(diskOptions =>
{
    diskOptions.AddDrive(driveName: "c:\\", 100);
    diskOptions.AddDrive(driveName: "f:\\", 300);
});
```

but `DiskStorageHealthCheck.CheckHealthAsync` implementation stops from the first non-compliance ([source code](https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/55a2915506fc2c9c3c8eda639913d55b795b1959/src/HealthChecks.System/DiskStorageHealthCheck.cs#L32)).

In the case where multiple disks have exceeded the threshold, only first one is reported to `ResponseWriter` provided to `UseHealthChecks`.

It would be desirable to be able to report each non-conformance (for exemple using multiple lines in the description) rather than let an administrator discover them in several tries.

This behavior would also be aligned with the general behavior where each registered `IHealthCheck` are checked regardless of the result of the previous one.

This could be made opt-in with a bool CheckAllDrives in `DiskStorageOptions`

I know I can use:
``` csharp
healthCheckBuilder.AddDiskStorageHealthCheck(
    setup: diskOptions =>diskOptions.AddDrive(driveName: "c:\\", 100),
    name: "diskstorage " +  "c:\\");

healthCheckBuilder.AddDiskStorageHealthCheck(
    setup: diskOptions =>diskOptions.AddDrive(driveName: "f:\\", 00),
    name: "diskstorage " +  "f:\\");
```
but the hierarchical approach seems to be more elegant than flat one.

**Comment by CarlosLanderas** [comment]
I agree with you. The main concern would be potencial big lists of failing checks being displayed in the UI.

Here is a sample with two failing disks:

![image](https://user-images.githubusercontent.com/16475649/71328089-71ac2400-2511-11ea-8584-4785ce27044c.png)

Disks are something "more controlled" as server don't use to have much of them, but talking about other king of checks the list could be huge.

What do you think?

**Comment by MuiBienCarlota** [comment]
@CarlosLanderas I agree with you. The "too big list" case should be taken into consideration rigth now.

A simple approach could be, in case of multiple failure, to provide a generic message (perhaps with an optional list of each failing driveNames) and using a hierarchical data to provide each individual failures {`driveName`, `minimunFreeMegabytes`, `ActualFreeMegabytes`}.

And because `Description` field is just a pretty message only for human use, it may be good to add to Data in any case (single and multiple AddDisk) an array of failure with {`driveName`, `minimunFreeMegabytes`, `ActualFreeMegabytes`}.

**Subsidiary question**: why are you using drive names ending with \ character like in `c:\`? Using this syntax may suggest that it is possible to use `c:\subfolder\`. This could be useful when using reparse point (ie if `subfolder` is in fact in another drive than `c:`). But actual implementation doesn't permits reparse point checks (equality with `DriveInfo.Name` received from `DriveInfo.GetDrives`). I think it could be better to use `c:` as main syntax for `driveName` field. To prevent breaking changes and to be compatible with `DriveInfo.GetDrives`, a trailing \ could be internally added if not present.

**Comment by unaizorrilla** [comment]
Hi @CarlosLanderas are you following this questions?

**Comment by sungam3r** [comment]
#958 allows to set custom description - short or long.

**Comment by sungam3r2022** [comment]
Fixed in #958
